### PR TITLE
Add issue and PR templates

### PR DESCRIPTION
**Pull request by natemcmaster**
cc @rowanmiller @divega 

cref https://github.com/blog/2111-issue-and-pull-request-templates

### CODE REVIEW

**Comment by smitpatel** [comment]
Should we have our customization to ask for model/dbcontext or query 😜

**Comment by natemcmaster** [comment]
@smitpatel I would expect a user to include those in "repro steps"

**Comment by laskoviymishka** [comment]
may it better to add some check list in PR and Issue template? like in 

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/aspnet/home/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

etc? there is awesome template for issue and PR in angular core repo. (see [ISSUE](https://github.com/angular/angular/blob/master/.github/ISSUE_TEMPLATE.md) and [PR](https://github.com/angular/angular/blob/master/.github/PULL_REQUEST_TEMPLATE.md))

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Summary of the changes
+ - Detail 1
+ - Detail 2
+
+Fixes #bugnumber
+
+You will need to sign a Contributor License Agreement before we can accept your pull request.
+Please review the guidelines for CONTRIBUTING.md for more details.
```
Comment (rowanmiller) [inline .github/PULL_REQUEST_TEMPLATE.md]: This shouldn't be needed as we have a bot that posts info when a CLA is needed.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+### Steps to reproduce
+Please include the minimal steps to reproduce the issue.
```
Comment (rowanmiller) [inline .github/ISSUE_TEMPLATE.md]: I would write something more like

> Ideally include a complete code listing that we can run to reproduce the issue. Alternatively, you can provide a project/solution that we can run.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+### Expected result
+
+
+### Actual result
+
```
Comment (rowanmiller) [inline .github/ISSUE_TEMPLATE.md]: I would combine these two, templates that have these (such as our Connect system) always get paired statements like "The entity is added"/"The entity is not added" - which is not useful. Maybe more like...

```
### The issue

Describe what is not working as expected. If you are seeing an exception, include the full exceptions details (message and stack trace).

**Comment by rowanmiller** [comment]
:shipit:

**Comment by natemcmaster** [comment]
Updated. @laskoviymishka good idea. I added a checklist.

**Comment by divega** [comment]
:+1:  

cc @ajcvickers and @anpete in case they have more insights.

**Pull request merged**